census/tensorflowcore/trainer/task.py

```diff
@@ -201,6 +202,16 @@ def run(target,
       for the outputed saved_model binary.
   """
 
+  # If job_dir_reuse is False then remove the job_dir if it exists
+  if not reuse_job_dir:
+    if tf.gfile.Exists(job_dir):
+      tf.gfile.DeleteRecursively(job_dir)
```

Comment:
@puneith : I'm very wary of doing deletes (especially equivalents to `rm -r`) in example code. We never know how someone is going to copy-paste. Maybe logging a warning or raising an exception would be preferable when the directory already exists and the `--reuse-job-dir` flag is not set?

Reply:
I appreciate that concern. I think given how many people have tripped over this issue (Kaggle, Gitbub, SO) *not* reusing the model dir as a default option covers the most common case for the sample use. Taking this sample and using this in production will need a similar use of the reuse pattern (may be default will be different). I am also logging the appropriate log messages and so user can always set the reuse to true. Raising the exception may not entirely help if user really wants to reuse the model dir. In any case this sample is optimizing for sample use of the code.

Reply:
My suggestion is to remove the `tf.gfile.DeleteRecursivelyCall` and instead raise an exception which asks them to manually delete if they desire to do so.

That way, if they want to reuse the directory, they would know to pass the `--reuse-job-dir` flag to bypass the exception and if they don't, they can manually go and delete the appropriate files/directories.

What do you think?

Reply:
So one thing is for sure, when someone runs a sample they would like to make sure it runs without any error. And my concern is if someone has to debug why they need to delete the directory when they don't really understand the context yet, they will scratch their head. And if they anyways follow the instruction of deleting the dir to get rid of the error then we might as well do that for them. What you are getting at is the software design aspect of this feature which is important but not something we are optimizing for upfront in the sample. When someone wants to adapt the sample to their production grade code they will need to study the code to make it work, but introducing a software design barrier early in the process of sample is not required.

Reply:
@puneith : Alright, that is reasonable. LGTM